### ISSUE TITLE
ggtext and parsing html with latest version of ggplot

### ISSUE THREAD

**Issue by Philip-Leftwich** [tidyverse/ggplot2#6752]
I found a problem with ggplot and rendering html text with ggtext or marquee

I expected to parse character string in html format and have the style implemented. 
This works for my code in ggplot 3.5.2 but not in ggplot 4.0.0

Here is the code to reproduce the bug:

``` r
library(ggplot2)
library(ggtext)

# Minimal dataframe
df <- data.frame(
  gRNA_type = rep(c("cd<sup><i>g338-384</i></sup>", 
                    "wt<sup><i>control</i></sup>"), each = 5),
  win = c(95, 88, 92, 90, 94, 45, 50, 48, 52, 47),
  loss = c(5, 12, 8, 10, 6, 55, 50, 52, 48, 53),
  id = 1:10
)

# Test plot
ggplot(df, aes(x = gRNA_type, y = win)) +
  geom_point() +
  theme_minimal() +
  theme(axis.text.x = element_markdown())
```

![](https://i.imgur.com/jckAdHS.png)

<sup>Created on 2025-11-19 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

<img width="861" height="638" alt="Image" src="https://github.com/user-attachments/assets/16aa35de-24cf-4f79-92ca-218946405f00" />

**Comment by teunbrand** [comment]
Thanks, I can confirm this on my end. The issue is element inheritance between S3 and S7 elements. `element_markdown()` isn't propagated to `axis.text.x.bottom` because `theme_minimal()$axis.text.x.bottom` already exists. The issue can be fixed by using `theme(axis.text.x.bottom = element_markdown())`.

**Comment by teunbrand** [comment]
I'm closing this issue as I'm assuming the thumbs-up means it was solved on your end. The path forward is for extensions to implement S7-based element extensions, which isn't really something ggplot2 can do for them.

**Comment by clauswilke** [comment]
@teunbrand I'm not sure I've followed all the recent changes. Is there something ggtext should do that it hasn't done yet?

**Comment by teunbrand** [comment]
The backward compatibility mechanisms in place mostly preserve how S3 elements work. The only thing it doesn't do is cross-inheritance between S3 and S7 theme elements. In a perfect world, extensions like ggtext would update to S7 elements. However, this would come at the cost of incrementing the ggplot2 version requirement. I can prepare a PR for ggtext and see where we want to go.

**Comment by clauswilke** [comment]
Sounds good, thanks. It looks to me like a sufficiently big regression that paying the price of increasing the minimum ggplot2 version is probably warranted. People can always install an earlier version of ggtext if they need to.

**Comment by teunbrand** [comment]
I found out that updating elements to S7 will not be sufficient.
This is how it should work: a child inherits from parent correctly.

``` r
library(ggplot2)

element_foo <- S7::new_class("element_foo", parent = element_text)

# Works
t <- theme_gray() +
  theme(axis.text.x = element_foo())

e <- calc_element("axis.text.x.bottom", t)
S7::S7_inherits(e, element_foo)
#> [1] TRUE
```

In the example below the `axis.text.x.bottom` is `element_text()`, so it will override the `element_foo()`. You may argue that the 'richer' class should take over the 'slender' class, but that is not how it works right now.

``` r
# Does not  work
t <- theme_minimal() +
  theme(axis.text.x = element_foo())

e <- calc_element("axis.text.x.bottom", t)
S7::S7_inherits(e, element_foo)
#> [1] FALSE
```

<sup>Created on 2025-11-26 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

The real change that broke this was in #6071 that bestowed `theme_minimal()` with a non-NULL `axis.text.x.bottom` element.
In any case, I'm reopening this issue because we could do with a little bit of thinking of what should happen here.
